src/Qowaiv/TelephoneNumber.cs

```diff
@@ -0,0 +1,138 @@
+﻿#pragma warning disable S1210
+// "Equals" and the comparison operators should be overridden when implementing "IComparable"
+// See README.md => Sortable
+
+#pragma warning disable S2328
+// "GetHashCode" should not reference mutable fields
+// See README.md => Hashing
+
+namespace Qowaiv
+{
+    using Qowaiv.Formatting;
+    using Qowaiv.Json;
+    using System;
+    using System.ComponentModel;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.Runtime.Serialization;
+    using System.Xml.Serialization;
+
+    /// <summary>Represents a telephone number.</summary>
+    [DebuggerDisplay("{DebuggerDisplay}")]
+    [Serializable]
+    [SingleValueObject(SingleValueStaticOptions.All, typeof(string))]
+    [OpenApiDataType(description: "telephone number", type: "TelephoneNumber", format: "TelephoneNumber")]
+    [TypeConverter(typeof(Conversion.TelephoneNumberTypeConverter))]
+    public partial struct TelephoneNumber : ISerializable, IXmlSerializable, IFormattable, IEquatable<TelephoneNumber>, IComparable, IComparable<TelephoneNumber>
+    {
+        /// <summary>Represents an empty/not set telephone number.</summary>
+        public static readonly TelephoneNumber Empty;
+
+        /// <summary>Represents an unknown (but set) telephone number.</summary>
+        public static readonly TelephoneNumber Unknown = new TelephoneNumber("?");
+
+        /// <summary>Gets the number of characters of telephone number.</summary>
+        public int Length => m_Value == null ? 0 : m_Value.Length;
+
+        /// <summary>Gets the type of the telephone number.</summary>
+        public TelephoneNumberType Type
+        {
+            get
+            {
+                if (IsEmpty())
+                {
+                    return TelephoneNumberType.None;
+                }
+                else if (IsUnknown())
+                {
+                    return TelephoneNumberType.Unknown;
+                }
+                else if (m_Value.StartsWith("+", StringComparison.InvariantCulture))
+                {
+                    return TelephoneNumberType.International;
+                }
+                else if (m_Value.StartsWith("0", StringComparison.InvariantCulture))
```

Comment:
If a telephone number starts with "00" it can also be an international number right, so shouldn't this check if the second is also a 0 or not and return International if so?

Reply:
00 can only be international, and will resolve into +.